### ISSUE TITLE
Household example simulation in docs fails to run

### ISSUE THREAD

**Issue by mikesmit** [PolicyEngine/policyengine.py#53]
1. clone the repo
2. create your venv
3. make install
4. load index.ipynb and run the first cell

Get the following error
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_9510/3061504181.py in <cell line: 3>()
      1 from policyengine import Simulation
      2
----> 3 sim = Simulation(
      4     country="uk",
      5     scope="household",
~/workspaces/policyengine.py/policyengine/simulation.py in __init__(self, country, scope, data, time_period, reform, baseline, verbose, options)
     67         self.country = country
     68         self.scope = scope
---> 69         self._set_dataset(data)
     70         self.time_period = time_period
     71         self.verbose = verbose
~/workspaces/policyengine.py/policyengine/simulation.py in _set_dataset(self, dataset)
     92
     93     def _set_dataset(self, dataset: str):
---> 94         if dataset in DATASETS[self.country]:
     95             self.data = DATASETS[self.country][dataset]
     96         elif dataset is None:
TypeError: unhashable type: 'dict'
```

**Comment by mikesmit** [comment]
TLDR: The declared types in Simulation are generally wrong and you can't use a dict as a key in a dict.

In this specific case this https://github.com/PolicyEngine/policyengine.py/blob/main/policyengine/simulation.py#L93-L99

```
    def _set_dataset(self, dataset: str):
        if dataset in DATASETS[self.country]:
            self.data = DATASETS[self.country][dataset]
        elif dataset is None:
            self.data = DEFAULT_DATASETS[self.country]
        else:
            self.data = dataset
```

code says "data" is an "str", but it can also be None or dict. In the case of the first example it's a dict:
```
from policyengine import Simulation

sim = Simulation(
    country="uk",
    scope="household",
    data={
        "employment_income": {
            2025: 30_000
        },
    },
    time_period=2025,
    reform={
        "gov.hmrc.income_tax.allowances.personal_allowance.amount": {
            "2025": 0,
        }
    }
)
sim.calculate("household/comparison/net_income_change")
```

so it tries to look it up in DASTASETS and fails.

**Comment by mikesmit** [comment]
I'm going to fix just this bug and then go and see if I can fix the types to be accurate (and address any issues that creates)

**Comment by mikesmit** [comment]
Waiting for contributor access so I can submit the first PR.
